maven/bnd-shared-maven-lib/src/main/java/aQute/bnd/maven/lib/configuration/BeanProperties.java

```diff
@@ -0,0 +1,89 @@
+package aQute.bnd.maven.lib.configuration;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+
+public class BeanProperties extends Properties {
+	private static final Pattern	KEY_P				= Pattern
+		.compile("(?<name>[^\\.\\[]+)(?:\\[(?<index>\\d+)\\])?\\.?");
+	private static final long serialVersionUID = 1L;
+	private final Logger			logger;
+
+	public BeanProperties(final Logger logger) {
```

Comment:
Why pass a Logger here? Should we follow normal convention and have this class get and use its own Logger?

Reply:
Yeah, good point. I moved the logic which was already passing the logger. I'll make the change.

Reply:
It was not passing a logger. It was just using the logger from the class is was part of.